src/Cli/Microsoft.DotNet.Cli.Utils/Muxer.cs

```diff
@@ -34,11 +34,32 @@ public string MuxerPath
 
         public Muxer()
         {
+            // Best-effort search for muxer.
+            // SDK sets DOTNET_HOST_PATH as absolute path to current dotnet executable
 #if NET6_0_OR_GREATER
-            _muxerPath = Environment.ProcessPath;
+            string processPath = Environment.ProcessPath;
 #else
-            _muxerPath = Process.GetCurrentProcess().MainModule.FileName;
+            string processPath = Process.GetCurrentProcess().MainModule.FileName;
 #endif
+            string hostName = "dotnet" + Constants.ExeSuffix;
+
+            // The current process should be dotnet in most normal scenarios except when dotnet.dll is loaded in a custom host like the testhost
+            if (!processPath.EndsWith(hostName, StringComparison.OrdinalIgnoreCase))
```

Comment:
This should fix the case where the process name is `notdotnet.exe`:

```suggestion
            if (!Path.GetFileNameWithoutExtension(processPath).Equals("dotnet", StringComparison.OrdinalIgnoreCase))
```